enhancements/ocp-coreos-layering/ocp-coreos-layering.md

```diff
@@ -0,0 +1,730 @@
+
+For now, updating kernel arguments will continue to happen via the MCD on each node via executing `rpm-ostree kargs` as it does today. See also https://github.com/ostreedev/ostree/issues/479
+
+#### Drain and reboot
+
+No changes in drain and reboot behavior, MCO will continue to perform drain and reboots like it does for existing releases.
+
+### Risks and Mitigations
+
+We're introducing a new format of shipping RHEL CoreOS image and allowing users to create their own custom image to override the default os on nodes.
+
+This brings in certain risk but hopefully with good documentation, cluster admins would know what they are doing and tests it before applying custom Image to cluster.
+
+### Drawbacks
+
+If it turns out that e.g. upgrading existing RHCOS systems in place is difficult, that will be a problem.
+
+## Design Details
+
+### Test Plan
+
+The FCOS / RHCOS base OCI containers will be published automatically by their respective CI pipelines. These pipelines already have a release promotion mechanism built into them. This means that early development versions of these containers are available (e.g. `:testing-devel` tags), hence other dependent systems, such as the MCO, can use them.
+
+For example, the MCO’s tests can use those containers to get a signal on potentially breaking changes to rpm-ostree as well as the base container images.
+
+MCO-specific tests will be run during each PR as part of the MCOs presubmits. Also, make sure that we're doing node scaling post-upgrade in some e2e tests.
+
+### Graduation Criteria
+
+ The goal for 4.12 is to have a supported path for users to apply hotfix (GA) into base RHEL CoreOS and ability to try out 3rd party software (Tech Preview).
+
+#### Dev Preview -> Tech Preview
+- `machine-os-content` is in the OSTree native container format and the MCO knows how to handle it.
+- A custom OS Image derived from base RHEL CoreOS image can be built by supplying Containerfile and can be supplied to MCO via MachineConfig `osImageURL` field to override the default. 
+- Installation of a new cluster is successful with the new OSTree native image format.
+- Upgrade of existing cluster is successful with the new format.
+- A CI test verifies the installation of a new cluster in this mode and installing an extension.
+- User documentation is available to create and apply custom os derived from RHEL CoreOS Image
+- Using layering, cluster admin can try installing specific 3rd party software on dev cluster and provide feedback.
+
+#### Tech Preview -> GA
+- All the goals listed above are implemented.
+- A supported model to apply hotfix on 4.12 based cluster is available
+
+#### Removing a deprecated feature
+
+Nothing will be deprecated as part of this.
+
+### Upgrade / Downgrade Strategy
+
+Since this change requires latest rpm-ostree version. We will backport required rpm-ostree version into latest OCP 4.11.z release. Any older cluster upgrade would go through that 4.11.z release pointed via CVO upgrade graph.
```

Comment:
Side note: this only affects the upgrade itself (need to pull new image to rebase to), but won't otherwise affect bootimage  issues which is tracked separately, right?

Reply:
right. Later is covered in `Handling booting old nodes` section.